ext/serializer.c

```diff
@@ -1256,6 +1256,25 @@ void ddtrace_serialize_span_to_array(ddtrace_span_data *span, zval *array) {
         }
 
         add_assoc_zval(el, "service", &prop_service_as_string);
+
+        // Add _dd.base_service if service name differs from mapped root service name
+        zval *prop_root_service = &span->root->property_service;
+        ZVAL_DEREF(prop_root_service);
+        zval prop_root_service_as_string;
+        ddtrace_convert_to_string(&prop_root_service_as_string, prop_root_service);
+
+        zval *new_root_name = zend_hash_find(service_mappings, Z_STR(prop_root_service_as_string));
+        if (new_root_name) {
+            zend_string_release(Z_STR(prop_root_service_as_string));
+            ZVAL_COPY(&prop_root_service_as_string, new_root_name);
+        }
+
+        if (strcasecmp(Z_STRVAL(prop_service_as_string), Z_STRVAL(prop_root_service_as_string)) != 0) {
```

Comment:
```suggestion
        if (zend_string_equals(Z_STR(prop_service_as_string), Z_STR(prop_root_service_as_string))) {
```
zend_string_equals is aware of the lengths directly and also does a pointer equality comparison (which likely will match if both are equal).

Reply:
👍 modified the suggestion a bit by using `zend_string_equals_ci` instead of `zend_string_equals` to keep the behavior consistent with `strcasecmp`

Reply:
Yeah, missed that it wasn't strcmp :-)